lib/core/versions/latest/DocumentComposer.ts

```diff
@@ -84,18 +84,7 @@ export default class DocumentComposer {
       }
 
       // Verify each service entry in array.
-      for (let serviceEntry of document.service) {
-        // 'type' is required and must be string type.
-        if (typeof serviceEntry.type !== 'string') {
-          throw new SidetreeError(ErrorCode.DocumentComposerServiceTypeNotString);
-        }
-
-        // 'serviceEndpoint' is required.
-        const serviceEntryType = typeof serviceEntry.serviceEndpoint;
-        if (serviceEntryType !== 'string' && serviceEntryType !== 'object') {
-          throw new SidetreeError(ErrorCode.DocumentComposerServiceEndpointMissingOrIncorrectType);
-        }
-      }
+      DocumentComposer.validateServiceEndpoints(document.service);
```

Comment:
Should be changed to `serviceEndpoints` according to the new paradigm (specified in protocol.md), basically we are decoupling our document state from the DID spec. Note that I have already changed `publicKey` to `publicKeys` in my recent PR.